crypto/src/keys.rs

```diff
@@ -139,6 +172,13 @@ static DIVERSIFY_GENERATOR_DOMAIN_SEP: Lazy<Fq> = Lazy::new(|| {
 });
 
 impl Diversifier {
+    /// Generate a new random diversifier.
+    pub fn generate<R: RngCore + CryptoRng>(mut rng: R) -> Self {
+        let mut diversifier = [0u8; DIVERSIFIER_LEN_BYTES];
+        rng.fill_bytes(&mut diversifier);
+        Diversifier(diversifier)
+    }
```

Comment:
This is good for now, but in the future I think we should probably shift the design to use a Poseidon-based diversifier design as in Orchard.